MONOREPO_ROADMAP.md

```diff
@@ -0,0 +1,383 @@
+
+##### Does:
+- Implement stores using Svelte's store contract (without referencing Svelte directly)
+  - Consider using a tool like [nanostores](https://github.com/nanostores/nanostores)
+- Provides everything needed to turn a Vite project into an [Evidence Application](#evidence-application)
+  - This does not include framework-specific ergonomics such as those in [@evidence-dev/lang-svelte](#evidence-devlang-svelte)
+
+##### Does Not:
+- Depend on Svelte, React, or any specific framework
+
+##### Exports:
+
+###### `@evidence-dev/sdk/config`
+- Utilities for reading and writing `evidence.config.yaml` according to a passed-in Zod Schema
+
+###### `@evidence-dev/sdk/lib`
+- Library code used across environments
+- `EvidenceError`, `isDebug`, `fillMissingData`, `setTrackProxy`, etc.
+
+###### `@evidence-dev/sdk/lib/datasources`
+- Library code to ease the creation of datasource plugins
+- Previously `@evidence-dev/db-commons`
+
+###### `@evidence-dev/sdk/plugins/components`
+- Utilities for loading [Component Plugins](#component-plugin)
+
+###### `@evidence-dev/sdk/plugins/datasources`
+- Utilities for loading [Datasource Plugins](#datasource-plugin)
+- Utilities for evaluating [Datasources](#datasources)
+
+###### `@evidence-dev/sdk/plugins/app-shells`
+- Exposes functions for setting up the [Application Shell](#application-shell) and copying the [Project](#evidence-project) into it
+
+###### `@evidence-dev/sdk/usql`
+- Library code for interacting with USQL
+- `Query`, `Metric`, etc.
+
+###### `@evidence-dev/sdk/logging`
+- Provide functions for error, warn, log, and debug
+- Provide hooks to integrate logs into Evidence UIs (e.g. a log panel in VS Code)
+  - These hooks should have a standard interface for the contents of the logs
+- Wrap / re-use an existing, tested logging framework to minimize first-party code
+  - e.g. `winston` or `pino` should be used under the hood
+
+
+#### `@evidence-dev/lang-svelte`
+
+Responsible for providing a smooth interface between the [Evidence SDK](#evidence-devsdk) and Svelte. 
+
+This functionality was previously provided by `@evidence-dev/component-utilities`, `@evidence-dev/preprocessor` and `@evidence-dev/sdk/utils/svelte`
+
+##### Does:
+- Provide a preprocessor to transform Evidence Markdown files to Svelte files
+  - Extracts & Bootstraps Queries
+  - Injects [Component Plugins](#component-plugin)
+  - Initializes [USQL](#evidence-devusql-duckdb)
+- Provide an interface to ease the use of [sdk](#evidence-devsdk) functionality in Svelte (e.g. using Svelte's context or lifecycle functions)
+  - Includes making SSR in an [Application Shell (plugin)](#application-shell-plugin) or [Evidence App](#evidence-application) trivial
+
+##### Does Not:
+- Copy files from an [Evidence Project](#evidence-project) to an [Application Shell](#application-shell)
+
+### UI Packages
+
```

Comment:
What about non-echarts charts?

Eg 
- Maps are all Leaflet.js
- VennDiagram is just css

Reply:
We briefly discussed Leaflet.js, and because it doesn't have any support for SSR (as far as I can tell), or rendering to SVG, it is being left out. The motivation here is to let our echarts configurations be used easily for server-rendering of charts or use in other frameworks